timemachine/potentials/potential.py

```diff
@@ -48,6 +48,9 @@ def __call__(self, conf: Conf, box: Optional[Box]) -> float | Array:
     def to_gpu(self, precision: Precision) -> "BoundGpuImplWrapper":
         return self.potential.to_gpu(precision).bind(np.asarray(self.params))
 
+    def __post_init__(self):
+        self.params = self.params.astype(np.float64).reshape(self.potential.params_shape())
```

Comment:
What is the intention of casting the parameters to float64? Do we want to silently convert parameters to float64?

Reply:
I think

1) all of our parameters on GPU implementation are 64 bit - regardless of `_f32` or `_f64`
2) up-casting should always be safe (but it can make things slower)

Reply:
We are already up-casting when we go to the GPU implementation. But we lose the original dtype that the potential was constructed with, which seems undesirable behavior. 

Also this potentially throws a bunch of warnings in jax if `self.params` is a jax array and 64 bit is not enabled.